crates/hyperqueue/src/worker/start.rs

```diff
@@ -354,6 +367,21 @@ async fn child_wait(
     }
 }
 
+#[inline(never)]
+fn check_error_filename(task_dir: TempDir) -> Option<DsError> {
+    let mut f = File::open(&get_custom_error_filename(&task_dir)).ok()?;
+    let mut buffer = [0; MAX_CUSTOM_ERROR_LENGTH];
+    let size = f
+        .read(&mut buffer)
+        .map_err(|e| log::debug!("Reading error file failed: {}", e))
+        .ok()?;
+    Some(if size == MAX_CUSTOM_ERROR_LENGTH {
+        DsError::GenericError("The task created a too long custom error message".to_string())
```

Comment:
In this case we should show the prefix/suffix along with this helper message. If the program dumps some long stacktrace or something, it's not good if the whole error is discarded.